### PR TITLE
Create aio-script.sh

### PR DESCRIPTION
**Pull request by codetronaut**
created a script to automate the installation of samba sharing server without login credentials.

### CODE REVIEW

**Comment by amanharitsh123** [comment]
@codetronaut 

- Can you please explain why you are downloading an additional config file for samba? 
- Why don't we just append an entry for the shared folder in the predefined samba config file?
- Please have a look at the Readme file, It is mentioned there that the script should work for both fedora and Debian based distros. Your script uses aptitude package manager, kindly make appropriate changes for Fedora as well. 
- Take user input for Folder to be shared and password.

**Comment by amanharitsh123** [comment]
- Please do a test run of the script before committing changes. You have made some blunder mistakes like missing space between "[" and "-z", spaces before and after "=".
- Ask for root permissions beforehand rather than using sudo.
- The script itself will do 4 stated tasks. It should accept options like -1,-2,-3,-4 to perform different tasks. So either program you permission and folder inputs accordingly as arguments or just take input from the user interactively after the script runs.
-  Check if the directory provided by the user actually exists or not.
- eth0 is not guaranteed to be the default ethernet interface. https://unix.stackexchange.com/questions/134483/why-is-my-ethernet-interface-called-enp0s10-instead-of-eth0
- Here is a solution for the same: https://github.com/amanharitsh123/sugarizer-school-box/blob/b9d3646d1f0e5abbe4bf4521aa030b35f3ec669b/image_scripts/hotspot.sh#L9
- Please mention the purpose of line 59 as a comment.

**Review by codetronaut — COMMENTED** [review]
Please review  the script.

**Comment by amanharitsh123** [comment]
Great Job, Options seems to work as expected. Some suggestions to work on further:

- Script throws an error for undefined function usage.
- Raise an error when the script has executed without specifying an option.
- Complete each function in different commit, it will be helpful to find bugs.
- Try to complete functionalities by 5 Feb before the coding period ends.

File: aio-script.sh [hunk 1]
```diff
@@ -24,6 +23,14 @@ function samba_config()
 
 function Update(){
 	echo "Updating..."
+if [ -f /etc/os-release ];
+    then
+        sudo aptitude update && sudo apt upgrade -y
```
Comment (amanharitsh123) [inline aio-script.sh]: Make it apt or apt-get.

**Comment by amanharitsh123** [comment]
Great. Merging it now.

**Pull request merged**